### DISCUSSION TITLE
Best practices for creating globally accessible shared values?

### DISCUSSION THREAD

**Discussion by jerryjfwang** [software-mansion/react-native-reanimated#4529] [Q&A] [answered]
I'm trying to make a shared value accessible globally. Is the best practice to use React Context? Is it possible to create shared values outside of React (eg: createSharedValue vs useSharedValue)? This would be very useful for working with state solutions like Zustand or Valtio.

Thanks in advance for the help!

**Comment by tomekzaw** [comment, marked as answer]
Hey @jerryjfwang

> Is the best practice to use React Context?

Yep, this should work.

> Is it possible to create shared values outside of React (eg: createSharedValue vs useSharedValue)?

Nope, this is not officially supported. Technically, you could use `makeMutable` but it's not a part of public API so we give no guarantees and things may break.

**Reply by XantreDev** [reply]
I think it would be great to have opportunities to create shared values outside of react

**Reply by ryskin** [reply]
just googled same, sad don't have

**Reply by MatiPl01** [reply]
I will just leave link to [makeMutable](https://docs.swmansion.com/react-native-reanimated/docs/advanced/makeMutable) docs we added recently. It might be helpful if you decide to use `makeMutable`.

**Comment by danieloi** [comment]
Context is giving me a mixed experience. Some components that hook into it, don't animate properly unless I save the file and trigger a fast refresh. It starts to work as expected afterwards @tomekzaw

**Reply by zgordon02** [reply]
I'm not experiencing anything of the sort. It triggering after a fast refresh tells me you have an effect that is refiring. Are you missing any deps in an effect hook?
